ethereum/networks/src/main/java/tech/pegasys/teku/networks/Eth2NetworkConfiguration.java

```diff
@@ -146,8 +146,14 @@ public String toString() {
     public Eth2NetworkConfiguration build() {
       checkNotNull(constants, "Missing constants");
 
+      final Spec spec = SpecFactory.create(constants);
+      // if the deposit contract was not set, default from constants
+      if (eth1DepositContractAddress.isEmpty()) {
+        eth1DepositContractAddress(
+            spec.getGenesisSpec().getConstants().getDepositContractAddress().toHexString());
+      }
       return new Eth2NetworkConfiguration(
```

Comment:
We should now be able to make `Eth2NetworkConfiguration.eth1DepositContractAddress` a simple `Eth1Address` rather than an `Optional<Eth1Address>`.  The constants parsing will fail if any field is missing, so we can rely on that value being available.

Reply:
ah ok i wasn't sure if the field wasn't required - that sounds like a good change to make

Reply:
raised #3730